server/src/main/java/org/bonitasoft/console/common/server/page/CustomPageService.java

```diff
@@ -120,13 +121,13 @@ public void ensurePageFolderIsUpToDate(final APISession apiSession, final PageRe
     }
 
     @SuppressWarnings("unchecked")
-    public Class<PageController> registerPage(final GroovyClassLoader pageClassLoader, final PageResourceProvider pageResourceProvider)
+    public Class<PageController> registerPage(final GroovyClassLoader pageClassLoader, final PageResourceProviderImpl pageResourceProvider)
```

Comment:
So I have update signature to use interface now.
However, there was some protected method in the old implementation that was use in the service that we do not want to expose in the api. That's why there is still some Impl visible in the code.